corehq/apps/hqcase/management/commands/ptop_preindex.py

```diff
@@ -85,13 +89,16 @@ def add_arguments(self, parser):
 
     def handle(self, **options):
         runs = []
-        all_es_indices = get_all_expected_es_indices()
+        all_es_indices = list(get_all_expected_es_indices())
         es = get_es_new()
         indices_needing_reindex = [info for info in all_es_indices if not es.indices.exists(info.index)]
```

Comment:
This seems like unnecessary work in the case of `--reset`. Could it (and the code below) be changed to something like this?
```py
if options['reset']:
    indices_needing_reindex = all_es_indices
else:
    indices_needing_reindex = [info for info in all_es_indices if not es.indices.exists(info.index)]
    if not indices_needing_reindex:
        print('Nothing needs to be reindexed')
        return
```